packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -943,20 +925,47 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 		}
 	}
 
-	public async getPendingBlobs(waitBlobsToAttach?: boolean): Promise<IPendingBlobs> {
-		if (waitBlobsToAttach) {
-			await this.stashPendingBlobs();
+	private async stashPendingBlobs(ids): Promise<void> {
+		for (const localId of ids) {
+			const entry = this.pendingBlobs.get(localId) as PendingBlob;
+			if (!entry.opsent) {
+				this.sendBlobAttachOp(localId, entry.storageId);
+			}
+			entry.handleP.resolve(this.getBlobHandle(localId));
 		}
+		return new Promise<void>((resolve) => {
+			if (this.allBlobsAttached) {
+				resolve();
+			} else {
+				this.once("allBlobsAttached", () => {
+					resolve();
+				});
+			}
+		});
+	}
+
+	public async getPendingBlobs(waitBlobsToAttach?: boolean): Promise<IPendingBlobs> {
 		const blobs = {};
-		for (const [key, entry] of this.pendingBlobs) {
-			blobs[key] = {
-				blob: bufferToString(entry.blob, "base64"),
-				storageId: entry.storageId,
-				attached: entry.attached,
-				acked: entry.acked,
-				minTTLInSeconds: entry.minTTLInSeconds,
-				uploadTime: entry.uploadTime,
-			};
+		const localIds = new Set<string>();
+		while (Object.keys(blobs).length < this.pendingBlobs.size) {
+			const ids = Array.from(this.pendingBlobs.keys()).filter(
+				(localId) => !localIds.has(localId),
+			);
+			if (waitBlobsToAttach) {
+				await this.stashPendingBlobs(ids);
```

Comment:
lets just inline the function here, and get rid of the loop inside it so something like 
```
for(const blob of this.pendingBlobs.filter((b)=>!localBlobs.has(b))){ 
... inlined functions
}
```

Reply:
Not sure what is localBlobs here

Reply:
its just localIds but storing instances, rather than ids

Reply:
added an example above